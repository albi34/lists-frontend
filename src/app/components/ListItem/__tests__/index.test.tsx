import React from 'react';
import { Store } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';
import { MemoryRouter } from 'react-router-dom';

import { ListItem } from '../index';

const renderList = (store: Store) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>
        <ListItem key="1" _id="1" name="test" />
      </MemoryRouter>
    </Provider>,
  );
};

describe('<ListItem />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore();
  });
  it('should match snapshot', () => {
    const ListItem = renderList(store);
    expect(ListItem.container.firstChild).toMatchSnapshot();
  });
});
