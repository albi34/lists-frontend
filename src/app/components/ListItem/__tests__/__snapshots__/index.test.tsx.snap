// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should match snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c0:active {
  opacity: 0.4;
}

<div>
  <a
    class="c0"
    href="/list/1"
  >
    &gt; 
    test
  </a>
   
  <button>
    X
  </button>
</div>
`;
